src/ast/CMakeLists.txt

```diff
@@ -58,6 +57,9 @@ if (STATIC_LINKING)
     target_link_libraries(ast "-Wl,--start-group" ${CLANG_EMBEDDED_CMAKE_TARGETS} ${LLVM_EMBEDDED_CMAKE_TARGETS} "-Wl,--end-group")
   else()
     llvm_map_components_to_libnames(llvm_libs ${llvm_lib_names})
+
+    find_package(LibClang_Static)
```

Comment:
This feels like a bug in your setup. Given that STATIC_LINKING changes the suffix to .a already, getting libclang.so here is making me think something is wrong with your prefix or your paths. 

Try `cmake --debug-find -DSTATIC_LINKING=ON .. |& less` to figure out where it's going wrong maybe?
Ultimately, we should not be hardcoding llvm internal names in this way and should be using the LLVM-Config.cmake macros instead. Maybe we need judicious use of llvm_config macro to do the right thing here?

Reply:
@BurntBrunch: in the initial revision I used `libclang_static` as the target name, which is defined in `ClangTargets.cmake`. However, this broke one of the embedded CI builds (llvm_clang, Alpine, llvm9) -- presumably because there's no `ClangTargets.cmake` in that setup (and llvm/clang libs are found in the default library search path).

I'm not very familiar with cmake internals, but the docs for [target_link_libraries](https://cmake.org/cmake/help/latest/command/target_link_libraries.html) suggest that the name should be a previously-defined cmake target, a plain name (e.g. `clang`) or a full path; not sure how it works with `libclang.a` as written here originally.

Based on the suffix being set to .a, I'd expect just using `clang` instead of `libclang.a` to work, but that didn't seem to be sufficient in my setup. Let me check again with `--debug-find`.

Reply:
> However, this broke one of the embedded CI builds (llvm_clang, Alpine, llvm9) -- presumably because there's no ClangTargets.cmake in that setup (and llvm/clang libs are found in the default library search path).

I checked the clang-dev-9.0.0-r1 package in alpine -- there's no libclang_static target in ClangTargets.cmake there (it must've been added after 9.0).